packages/js/navigation/src/index.js

```diff
@@ -178,6 +183,28 @@ export function getQuery() {
 	return {};
 }
 
+/**
+ * Like getQuery but in useHook format for easy usage in React functional components
+ *
+ * @return {Record<string, string>} Current query object, defaults to empty object.
+ */
+export const useQuery = () => {
+	const [ queryState, setQueryState ] = useState( {} );
+	const [ locationChanged, setLocationChanged ] = useState( false );
+	useLayoutEffect( () => {
+		return addHistoryListener( () => {
+			setLocationChanged( true );
+		} );
+	}, [] );
+
+	useEffect( () => {
+		const query = getQuery();
```

Comment:
Should we check if the `locationChanged` is true before updating the query state?  It seems that this effect will be triggered again when we change the state in line 203.

But if we add a check, we might also need to change the default state to true so that we can get the query state at the first time. Or is it possible to set default queryState to `getQuery()`?

Reply:
nice catch! I think this should work:

https://user-images.githubusercontent.com/27843274/182795502-84a7f68d-2b80-4aad-bfbf-f946f4278043.mp4